SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -363,6 +366,143 @@ void InterestFactory::AddServerGameplayDebuggerCategoryReplicatorActorInterest(I
 }
 #endif
 
+bool UnrealServerInterestFactory::CreateClientInterestDiff(const AActor* InActor, Worker_CommandRequest& Request,
+														   const bool bOverwrite) const
+{
+	const APlayerController* PlayerController = Cast<APlayerController>(InActor);
+	if (!ensure(GetDefault<USpatialGDKSettings>()->bUseClientEntityInterestQueries)
+		|| !ensureMsgf(PlayerController != nullptr, TEXT("Only PCs can use client entity interest: %s"), *GetNameSafe(InActor)))
+	{
+		return false;
+	}
+
+	// If and when the Runtime has a better interest changing API, we can ask for a diff here instead.
+	TArray<Worker_EntityId> ClientInterestedEntities = GetClientInterestedEntityIds(PlayerController);
+
+	USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(PlayerController->GetNetDriver());
+
+	// MAKE DEBUGGING EASIER, REVERT ME
+#if WITH_EDITOR
+	ClientInterestedEntities.Sort();
+#endif
+
+	static ChangeInterestRequest ChangeInterestRequestData;
+	ChangeInterestRequestData.Clear();
+
+	if (USpatialNetConnection* NetConnection = Cast<USpatialNetConnection>(PlayerController->NetConnection))
+	{
+		// Add non-auth interest
+		TSet<Worker_EntityId_Key> FullInterested;
+		FullInterested.Reserve(ClientInterestedEntities.Num());
+		for (auto EntityId : ClientInterestedEntities)
+		{
+			FullInterested.Add(EntityId);
+		}
+		TSet<Worker_EntityId_Key> Add = FullInterested.Difference(NetConnection->EntityInterestCache);
+		TSet<Worker_EntityId_Key> Remove = NetConnection->EntityInterestCache.Difference(FullInterested);
+		NetConnection->EntityInterestCache = FullInterested;
+
+		if (bOverwrite)
+		{
+			Add = FullInterested;
+			Remove.Empty();
+		}
```

Comment:
NIT if bOverwrite is true this should be done before the diffing?

Reply:
Functionality is the same, but I guess you'd save on the diff computations